File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int summ(TreeNode* root,int h,int i){
        if(root==NULL) return 0;
        
        if(i==h)
            return root->val;
        
        int lsum=summ(root->left,h,i+1);
        int rsum=summ(root->right,h,i+1);
        
        return lsum+rsum;
        
    }
    int height(TreeNode* root){
        if(root == NULL) return 0;
        
        int lh=height(root->left);
        int rh=height(root->right);
        
        return max(lh,rh)+1;
    }
    int deepestLeavesSum(TreeNode* root) {
        int h=height(root);
        
        
        return summ(root,h,1);
    }        

};

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        int sum = 0;
        while(!q.isEmpty()){
            int n = q.size();
            sum=0;
            for(int i = 0;i<n;i++){
                TreeNode n1 = q.poll();
                sum+=n1.val;
                if(n1.left!=null)
                    q.add(n1.left);
                if(n1.right!=null)
                    q.add(n1.right);
            }
        }
        return sum;
    }
}